core/src/main/java/org/infinispan/notifications/impl/AbstractListenerImpl.java

```diff
@@ -179,7 +181,15 @@ protected boolean validateAndAddListenerInvocations(Object listener, AbstractInv
       Listener l = testListenerClassValidity(listener.getClass());
       boolean foundMethods = false;
       builder.setTarget(listener);
-      builder.setSubject(Security.getSubject());
+
+      Subject subject;
+      if (System.getSecurityManager() == null) {
+         subject = Security.getSubject();
```

Comment:
That's in fact "lightweight" security. It has no pretense of protecting you from malicious code, but it is much faster than a SecurityManager :) If one is present it will fallback to that.